js/src/attribute-mapping/attribute-mapping-rule-modal.js

```diff
@@ -241,7 +237,9 @@ const AttributeMappingRuleModal = ( { rule, onRequestClose = noop } ) => {
 							onCategoriesChange={ ( categories ) => {
 								updateRule( {
 									...newRule,
-									categories: categories.join( ',' ),
+									categories: categories
+										.map( ( category ) => category.value )
```

Comment:
WDYT of moving the mapping inside `AttributeMappingCategoryControl`? In this way, the component asks for a string array on `selectedCategories` prop and the callback value of `onCategoriesChange` could be a string array as well, which has a consistent data shape on the component APIs.